File: sdk/sdk-api-java/src/main/java/com/xpay/sdk/api/entity/FileRequest.java
package com.xpay.sdk.api.entity;

import java.util.List;
import java.util.Map;
import java.util.TreeMap;

public class FileRequest {
    private String method;
    private String version;
    private String randStr;
    private String signType;
    private String mchNo;
    private String secKey;
    private String timestamp; //时间戳，精确倒毫秒
    private String extras; //额外的参数(如有需要，可以使用)
    private String hash; //文件的hash值，如：md5值
    private List<FileInfo> files;

    public String getMethod() {
        return method;
    }

    public void setMethod(String method) {
        this.method = method;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public String getRandStr() {
        return randStr;
    }

    public void setRandStr(String randStr) {
        this.randStr = randStr;
    }

    public String getSignType() {
        return signType;
    }

    public void setSignType(String signType) {
        this.signType = signType;
    }

    public String getMchNo() {
        return mchNo;
    }

    public void setMchNo(String mchNo) {
        this.mchNo = mchNo;
    }

    public String getSecKey() {
        return secKey;
    }

    public void setSecKey(String secKey) {
        this.secKey = secKey;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public String getHash() {
        return hash;
    }

    public void setHash(String hash) {
        this.hash = hash;
    }

    public String getExtras() {
        return extras;
    }

    public void setExtras(String extras) {
        this.extras = extras;
    }

    public List<FileInfo> getFiles() {
        return files;
    }

    public void setFiles(List<FileInfo> files) {
        this.files = files;
    }

    public Map<String, String> getTextBodySort(){
        Map<String, String> textParam = new TreeMap<>();
        textParam.put("method", getMethod());
        textParam.put("version", getVersion());
        textParam.put("randStr", getRandStr());
        textParam.put("signType", getSignType());
        textParam.put("mchNo", getMchNo());
        textParam.put("secKey", getSecKey());
        textParam.put("timestamp", getTimestamp());
        textParam.put("extras", getExtras());
        textParam.put("hash", getHash());
        return textParam;
    }
}

File: sdk/sdk-api-php/entity/FileRequest.php
<?php
namespace xpay;

class FileRequest implements \JsonSerializable {
    private string $method  = '';
    private string $version  = '';
    private string $randStr  = '';
    private string $signType  = '';
    private string $mchNo  = '';
    private string $secKey = '';
    private string $timestamp = ''; //时间戳，精确倒毫秒
    private string $extras = ''; //额外的参数(如有需要，可以使用)
    private string $hash = ''; //文件的hash值，如：md5值，多个文件时用英文的逗号分割

    /**
     * 文件数组列表
     * @var FileInfo[]
     */
    private array $fileInfos = [];

    /**
     * 进行json序列化时返回的内容
     * @return array|mixed
     */
    public function jsonSerialize()
    {
        $vars = get_object_vars($this);
        return $vars;
    }

    /**
     * @return string
     */
    public function getMethod(): string
    {
        return $this->method;
    }

    /**
     * @param string $method
     */
    public function setMethod(string $method): void
    {
        $this->method = $method;
    }

    /**
     * @return string
     */
    public function getVersion(): string
    {
        return $this->version;
    }

    /**
     * @param string $version
     */
    public function setVersion(string $version): void
    {
        $this->version = $version;
    }

    /**
     * @return string
     */
    public function getRandStr(): string
    {
        return $this->randStr;
    }

    /**
     * @param string $randStr
     */
    public function setRandStr(string $randStr): void
    {
        $this->randStr = $randStr;
    }

    /**
     * @return string
     */
    public function getSignType(): string
    {
        return $this->signType;
    }

    /**
     * @param string $signType
     */
    public function setSignType(string $signType): void
    {
        $this->signType = $signType;
    }

    /**
     * @return string
     */
    public function getMchNo(): string
    {
        return $this->mchNo;
    }

    /**
     * @param string $mchNo
     */
    public function setMchNo(string $mchNo): void
    {
        $this->mchNo = $mchNo;
    }

    /**
     * @return string
     */
    public function getSecKey(): string
    {
        return $this->secKey;
    }

    /**
     * @param string $secKey
     */
    public function setSecKey(string $secKey): void
    {
        $this->secKey = $secKey;
    }

    /**
     * @return string
     */
    public function getTimestamp(): string
    {
        return $this->timestamp;
    }

    /**
     * @param string $timestamp
     */
    public function setTimestamp(string $timestamp): void
    {
        $this->timestamp = $timestamp;
    }

    /**
     * @return string
     */
    public function getExtras(): string
    {
        return $this->extras;
    }

    /**
     * @param string $extras
     */
    public function setExtras(string $extras): void
    {
        $this->extras = $extras;
    }

    /**
     * @return string
     */
    public function getHash(): string
    {
        return $this->hash;
    }

    /**
     * @param string $hash
     */
    public function setHash(string $hash): void
    {
        $this->hash = $hash;
    }

    /**
     * @return FileInfo[]
     */
    public function getFileInfos(): array
    {
        return $this->fileInfos;
    }

    /**
     * @param FileInfo[]
     */
    public function setFileInfos(array $fileInfos): void
    {
        $this->fileInfos = $fileInfos;
    }

    /**
     * @return array
     */
    public function getTextBodySorted(): array
    {
        //通过反射取得所有属性和属性的值
        $reflect = new \ReflectionClass(self::class);
        $props = $reflect->getProperties(\ReflectionProperty::IS_PUBLIC | \ReflectionProperty::IS_PRIVATE | \ReflectionProperty::IS_PROTECTED);
        $bodyArr = [];
        foreach ($props as $prop) {
            $prop->setAccessible(true);
            $key = $prop->getName();
            $value = $prop->getValue($this);

            if($key === 'fileInfos'){
                continue;
            }

            $bodyArr[$key] = $value;
        }

        //按key的字典序升序排序，并保留key值
        ksort($bodyArr);
        return $bodyArr;
    }
}
